lib/modules/datasource/bazel/index.ts

```diff
@@ -0,0 +1,65 @@
+import is from '@sindresorhus/is';
+import { ExternalHostError } from '../../../types/errors/external-host-error';
+import { cache } from '../../../util/cache/package/decorator';
+import { HttpError } from '../../../util/http';
+import { joinUrlParts } from '../../../util/url';
+import { BzlmodVersion } from '../../versioning/bazel-module/bzlmod-version';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { BazelModuleMetadata } from './schema';
+
+export class BazelDatasource extends Datasource {
+  static readonly id = 'bazel';
+
+  static readonly bazelCentralRepoUrl =
+    'https://raw.githubusercontent.com/bazelbuild/bazel-central-registry/main';
+
+  override readonly defaultRegistryUrls = [BazelDatasource.bazelCentralRepoUrl];
+  override readonly customRegistrySupport = true;
+  override readonly caching = true;
+
+  static packageMetadataPath(packageName: string): string {
+    return `/modules/${packageName}/metadata.json`;
+  }
+
+  constructor() {
+    super(BazelDatasource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${BazelDatasource.id}`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl!}:${packageName}`,
+  })
+  async getReleases({
+    registryUrl,
+    packageName,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const path = BazelDatasource.packageMetadataPath(packageName);
+    const url = joinUrlParts(registryUrl!, path);
+
+    const result: ReleaseResult = { releases: [] };
+    try {
+      const { body: metadata } = await this.http.getJson(
+        url,
+        BazelModuleMetadata
+      );
+      result.releases = metadata.versions
+        .filter((v) => is.falsy(metadata.yanked_versions[v]))
```

Comment:
is yanked like deprecated? so we should return the version but mark as deprecated?

https://github.com/renovatebot/renovate/blob/6ec865be62fa9597e9027eec69a6cddd5def70e5/lib/modules/datasource/types.ts#L47

that way renovate will know the version and tried to up- or downgrade to a non deprecated version.

Reply:
@viceice It does mean that it is deprecated. I will update the `Release` creation. If it is not deprecated, do you want the `isDeprecated` value to appear as `false` or not include it?

Reply:
For now, I did not include `isDeprecated` if the value would be `false`.